lms/djangoapps/instructor/views/data_access.py

```diff
@@ -0,0 +1,348 @@
+            bulk_queries.append(row)
+        QueriesStudents.objects.bulk_create(bulk_queries)
+        QueriesTemporary.objects.filter(id=temp_query.id).update(done=True)  # pylint: disable=no-member
+        if associate_group is not None:
+            grouped_temp = GroupedTempQueriesSubqueries(grouped_id=associate_group,
+                query_id=temp_query.id)  # pylint: disable=no-member
+            grouped_temp.save()
+
+    except Exception as ex:
+        QueriesTemporary.objects.filter(id=temp_query.id).update(done=None)  # pylint: disable=no-member
+        raise(ex)
+
+    #on every 10th query, purge the temporary queries
+    rand = random.random()
+    if rand > .9:
+        purge_temporary_queries()
+    return {temp_query.id: students}  # pylint: disable=no-member
+
+
+def purge_temporary_queries():
+    """
+    Delete queries made more than 30 minutes ago along with the saved students from those queries
+    """
+    minutes30ago = datetime.datetime.now() - datetime.timedelta(minutes=30)
+    old_queries = QueriesTemporary.objects.filter(created__lt=minutes30ago)
+    saved_students = QueriesStudents.objects.filter(query_id__in=old_queries.values_list(DATABASE_FIELDS.ID))
+    saved_students.delete()
+    old_queries.delete()
+
+
+@task  # pylint: disable=E1102
+def make_total_query(existing_queries):
+    """
+    Given individual queries that have already been made , aggregate students associated with those queries
+    """
+    aggregate_existing = set()
+    if len(existing_queries) != 0:
+        queryset = make_existing_query(existing_queries).values_list(DATABASE_FIELDS.ID,
+            DATABASE_FIELDS.EMAIL,
+            DATABASE_FIELDS.PROFILE_NAME).distinct()
+        for row in queryset:
+            aggregate_existing.add((row[0], row[1], row[2]))
+    return aggregate_existing
+
+
+def get_problem_users(course_id, query):
+    """
+    Gets Students filtered on specified problem-specific query criteria
+    """
+    if query.filter == PROBLEM_FILTERS.OPENED:
+        results = open_query(course_id, query)
+    elif query.filter == PROBLEM_FILTERS.COMPLETED:
+        results = completed_query(course_id, query)
+    elif query.filter == PROBLEM_FILTERS.NOT_OPENED:
+        results = not_open_query(course_id, query)
+    elif query.filter == PROBLEM_FILTERS.NOT_COMPLETED:
+        results = not_completed_query(course_id, query)
+    return results
+
+
+def get_section_users(course_id, query):
+    """
+    Gets Students filtered on specified section-specific query criteria
+    """
+    if query.filter == SECTION_FILTERS.OPENED:
+        results = open_query(course_id, query)
+    elif query.filter == SECTION_FILTERS.NOT_OPENED:
+        results = not_open_query(course_id, query)
+    return results
+
+
+def make_existing_query(existing_queries):
+    """
+    Aggregates single queries in a group into one unified set of students
+    """
+    query = User.objects
+    if existing_queries is None:
```

Comment:
I need to return User.objects if existing_queries is None and if it isn't use the "query" later on